neutron_vpnaas/services/vpn/device_drivers/meter_ipsec.py

```diff
@@ -0,0 +1,336 @@
+#    under the License.
+import time
+import os
+import netaddr
+import traceback
+from neutron.agent.linux import iptables_manager
+from neutron.common import rpc as n_rpc
+from neutron import context
+from oslo_config import cfg
+from oslo_log import log as logging
+from oslo_service import loopingcall
+
+from neutron_vpnaas.services.vpn.device_drivers import ipsec
+
+LOG = logging.getLogger(__name__)
+
+meter_opts = [
+    cfg.BoolOpt('vpn_meter_enable',
+                default=False,
+		help=_('Enable flag for VPN metering function')),
+    cfg.IntOpt('vpn_measure_interval',
+               default=30,
+	       help=_('The interval between two metering measures')),
+    cfg.IntOpt('vpn_report_interval',
+	       default=300,
+	       help=_('The interval between two metering reports'))
+]
+
+cfg.CONF.register_opts(meter_opts, 'meter')
+
+WRAP_NAME = 'neutron-vpnmeter'
+TOP_CHAIN = WRAP_NAME + '-FORWARD'
+RULE_OUT = '-o-'
+RULE_IN = '-i-'
+
+MAX_CHAIN_LEN_WRAP = 11
+MAX_CHAIN_LEN_NOWRAP = 28
+
+def get_chain_name(chain_name, wrap=True):
+    if wrap:
+        return chain_name[:MAX_CHAIN_LEN_WRAP]
+    else:
+        return chain_name[:MAX_CHAIN_LEN_NOWRAP]
+
+
+class MeterProcess(object):
+    def __init__(self, conf, process_id, vpnservice, namespace):
+        if self.conf.meter.vpn_meter_enable:
+            self.meter_iptables_manager = iptables_manager.IptablesManager(
+	        state_less=True,
+	        namespace=namespace,
+	        binary_name='neutron-vpnmeter',
+	        use_ipv6=False)
+    
+    def conn_id_check(self, vpnservice):
+       #check whether the conn_id is existing before, it yes, just empty the new rules and delete the old ones
+       	config_file = os.path.join(self.etc_dir, 'ipsec.conf')
+	conn_ids = []
+
+	if os.path.isfile(config_file):
+	    ipsec_conns = vpnservice['ipsec_site_connections']
+	    fp = open(config_file, 'r')
+	    lines = fp.readlines()
+	    fp.close()
+	    for line in lines:
+	        if line.find('conn ') == -1 or line.find('%default') != -1:
+		    continue
+		conn_ids.append(line.split()[1])
+	    
+	    for ipsec_conn in ipsec_conns:
+	        if ipsec_conn['id'] in conn_ids:
+		    conn_ids.remove(ipsec_conn['id'])
+         
+	return conn_ids
+
+
+class MeterIPsecDriver(ipsec.IPsecDriver):
+
+    def __init__(self, vpnservice, host):
+       super(MeterIPsecDriver, self).__init__(vpnservice, host)
+       if self.conf.meter.vpn_meter_enable:
+           measure_interval = self.conf.meter.vpn_measure_interval
+	   self.metering_loop = loopingcall.FixedIntervalLoopingCall(
+	       self._metering_loop
+	   )
+	   self.last_report = 0
+	   self.metering_loop.start(interval=measure_interval)
+	   self.tenant_conn_ids = {}
+           self.conn_tenant_id = {}
+           self.metering_out_infos = {}
+	   self.metering_in_infos = {}
+	   self.metering_infos = [self.metering_out_infos, self.metering_in_infos]
+
+    def destroy_process(self, process_id):
```

Comment:
process disable related operations should handled by Ipsec drivers, why you disable here?
do you really need override destroy_process?